mesonbuild/compilers/mixins/clike.py

```diff
@@ -465,6 +465,14 @@ def build_wrapper_args(self, env: 'Environment',
         if self.linker_to_compiler_args([]) == ['/link'] and largs != [] and '/link' not in extra_args:
             extra_args += ['/link']
 
+        # for MSVC, all linker flags must be converted now
+        # otherwise the reordering of arglist will apply
+        # and -L flags will be reordered into breaking form
+        #
+        # See arglist._should_prepend
+        if self.linker_to_compiler_args([]) == ['/link']:
```

Comment:
I wonder if a better way to check is `self.get_argument_syntax() == 'msvc'`?

Reply:
Makes sense, but it would be stylistically inconsistent with your own check style, just above this block: https://github.com/mesonbuild/meson/blob/b2473b61cc2c61218f5be4057af0e42b63d00048/mesonbuild/compilers/mixins/clike.py#L464-L467

Reply:
I just refactored the highlighted block, letting me merge the two checks together.